internal/session/types.go

```diff
@@ -36,6 +36,8 @@ type UserSession struct {
 	KeepMeLoggedIn      bool
 	AuthenticationLevel authentication.Level
 	LastActivity        int64
+	FirstFactorAuthn    int64
```

Comment:
I would append Timestamp. Something like FirstFactorAuthnTimestamp and same for the second factor timestamp.